addons/sale_project/models/sale_order.py

```diff
@@ -168,3 +168,15 @@ def _prepare_analytic_account_data(self, prefix=None):
         result = super(SaleOrder, self)._prepare_analytic_account_data(prefix=prefix)
         result['plan_id'] = self.company_id.analytic_plan_id.id or result['plan_id']
         return result
+    def _get_plan(self):
+        so_line_new_project = self.order_line.filtered(lambda sol: sol.is_service and sol.product_id.service_tracking in ['project_only', 'task_in_project'])
+        if so_line_new_project:
+            plan = self.env['account.analytic.plan'].sudo().search([('company_id', '=', False)], limit=1)
```

Comment:
Yes, having no company id on the plan is something we want. The goal is that the project created from an SO should not have a company_id, unless the SO already has an account linked to it, and that this account has a company_id set. 
I did not think about the default plan of the setting though. I updated the code a bit in order to take it into account too. If the default plan has no company_id, then we follow the rule above, otherwise, we have to propagate the company from the plan to the account and the project.